pkg/core/application_store.go

```diff
@@ -0,0 +1,54 @@
+package core
+
+import (
+	"context"
+
+	"github.com/fuseml/fuseml-core/gen/application"
+)
+
+// ApplicationStore describes in memory store for applications
+type ApplicationStore struct {
+	items map[string]*application.Application
+}
+
+// NewApplicationStore returns an in-memory application store instance
+func NewApplicationStore() *ApplicationStore {
+	return &ApplicationStore{
+		items: make(map[string]*application.Application),
+	}
+}
+
+// Find returns a application identified by id
+func (as *ApplicationStore) Find(ctx context.Context, name string) *application.Application {
+	return as.items[name]
+}
+
+// GetAll returns all applications of a given type.
+// If type is not specified, return all applications.
+func (as *ApplicationStore) GetAll(ctx context.Context, applicationType *string, applicationWorkflow *string) ([]*application.Application, error) {
+	result := make([]*application.Application, 0, len(as.items))
+	for _, app := range as.items {
+		if applicationWorkflow != nil && app.Workflow != *applicationWorkflow {
+			continue
+		}
+		if applicationType != nil && app.Type != *applicationType {
+			continue
+		}
+		result = append(result, app)
+	}
+	return result, nil
+}
+
+// Add adds a new application, based on the Application structure provided as argument
+func (as *ApplicationStore) Add(ctx context.Context, a *application.Application) (*application.Application, error) {
+	as.items[a.Name] = a
```

Comment:
Adding an application with the same name results in the previous one being overwritten. Is that intended ? Or should this function return an error if the name is already taken ?

Reply:
Hm good question. As application names are currently same as workflow names, let me check how it is done for workflows...

Reply:
... yep, it's just the same replace as here. I wonder where we should add a check for possible conflict. Probably at both places, but the user facing one is currently the workflow part, applications are created by running workflow (so if the conflict is avoided at workflow level, it should not exist on application one).

Reply:
I think we can write this off as intended behavior, if you're expecting subsequent calls to add the same application again every time there's a workflow run updating the application.